master/buildbot/status/status_nats.py

```diff
@@ -0,0 +1,148 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+
+"""Push events to nats
+
+."""
+
+from buildbot.status.status_push import StatusPush
+from buildbot.status.persistent_queue import DiskQueue, MemoryQueue, PersistentQueue
+from buildbot import config
+
+from twisted.python import log
+import pynats
+import urlparse
+import json
+
+
+class NatsStatusPush(StatusPush):
+    """Event streamer to a Nats server."""
+
+    def __init__(self, serverUrl, subject="katana", debug=None, maxMemoryItems=None,
+                 maxDiskItems=None, chunkSize=200, maxPushSize=2 ** 20, extra_post_params=None, **kwargs):
+        """
+        @serverUrl: The Nats server to be used to push events notifications to.
+        @subject: The subject to use when publishing data
+        @maxMemoryItems: Maximum number of items to keep queued in memory.
+        @maxDiskItems: Maximum number of items to buffer to disk, if 0, doesn't
+        use disk at all.
+        @debug: Save the json with nice formatting.
+        @chunkSize: maximum number of items to send in each at each PUSH.
+        @maxPushSize: limits the size of encoded data for AE, the default
+        is 1MB.
+        """
+        if not serverUrl:
+            raise config.ConfigErrors(['NatsStatusPush requires a serverUrl'])
+
+        # Parameters.
+        self.serverUrl = serverUrl
+        self.extra_post_params = extra_post_params or {}
+        self.debug = debug
+        self.chunkSize = chunkSize
+        self.lastPushWasSuccessful = True
+        self.subject = subject
+        self.maxPushSize = maxPushSize
+        self.c = None
+
+        if maxDiskItems != 0:
+            # The queue directory is determined by the server url.
+            path = ('events_' +
+                    urlparse.urlparse(self.serverUrl)[1].split(':')[0])
+            queue = PersistentQueue(
+                        primaryQueue=MemoryQueue(maxItems=maxMemoryItems),
+                        secondaryQueue=DiskQueue(path, maxItems=maxDiskItems))
+        else:
+            path = None
+            queue = MemoryQueue(maxItems=maxMemoryItems)
+
+        # Use the unbounded method.
+        StatusPush.__init__(self, serverPushCb=NatsStatusPush.pushData,
+                            queue=queue, path=path, **kwargs)
+
+    def wasLastPushSuccessful(self):
```

Comment:
This is necessary mainly because the super class expect it, i didn't make that super class so I don't know why that decision was made